From 94bdc4e055d10b89c7683c7966791612a7aed455 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 30 Jul 2024 17:45:33 +0300
Subject: [PATCH 12/13] plat: lx2160a: fix building without
 NXP_NV_SW_MAINT_LAST_EXEC_DATA

Fix compiler errors encountered when disabling non-volatile storage of
execution state (NXP_NV_SW_MAINT_LAST_EXEC_DATA := no) while keeping
watchdog restart enabled (NXP_WDOG_RESTART := yes).

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/nxp/ddr/phy-gen2/phy.h | 2 +-
 plat/nxp/soc-lx2160a/soc.c     | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/nxp/ddr/phy-gen2/phy.h b/drivers/nxp/ddr/phy-gen2/phy.h
index 5e80f3638..7ddc7c7c6 100644
--- a/drivers/nxp/ddr/phy-gen2/phy.h
+++ b/drivers/nxp/ddr/phy-gen2/phy.h
@@ -6,7 +6,7 @@
 #if !defined(PHY_H) && defined(NXP_WARM_BOOT)
 #define PHY_H
 
-#include <flash_info.h>
+#include <drivers/nxp/flexspi/flash_info.h>
 
 /* To store sector size to be erase on flash*/
 #define PHY_ERASE_SIZE F_SECTOR_ERASE_SZ
diff --git a/plat/nxp/soc-lx2160a/soc.c b/plat/nxp/soc-lx2160a/soc.c
index 137c24ef8..4aebb1393 100644
--- a/plat/nxp/soc-lx2160a/soc.c
+++ b/plat/nxp/soc-lx2160a/soc.c
@@ -521,10 +521,12 @@ void soc_init(void)
 static uint64_t wdog_interrupt_handler(uint32_t id, uint32_t flags,
 					  void *handle, void *cookie)
 {
+#ifdef NXP_NV_SW_MAINT_LAST_EXEC_DATA
 	uint8_t data = WDOG_RESET_FLAG;
 
 	wr_nv_app_data(WDT_RESET_FLAG_OFFSET,
 		       (uint8_t *)&data, sizeof(data));
+#endif
 
 	mmio_write_32(NXP_RST_ADDR + RSTCNTL_OFFSET, SW_RST_REQ_INIT);
 
-- 
2.43.0

